include/revng/AutoEnforcer/AutoEnforcerLibraryRegistry.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+
+#include "revng/AutoEnforcer/PipelineLoader.h"
+
+namespace AutoEnforcer {
+
+class AutoEnforcerLibraryRegistry {
+public:
+  AutoEnforcerLibraryRegistry() { getInstances().push_back(this); }
+
+  static void registerAllContainersAndEnforcers(PipelineLoader &Loader) {
+    for (const auto &Reg : getInstances())
+      Reg->registerContainersAndEnforcers(Loader);
+  }
+
+  static KindsRegisty registerAllKinds() {
+    KindsRegisty Registry;
+    for (const auto &Reg : getInstances())
+      Reg->registerKinds(Registry);
+
+    for (auto &Kind : Registry) {
+      Kind.getRootAncestor()->assign();
+      Kind.Granularity->getRootAncestor()->assign();
+    }
+    return Registry;
+  }
+
+  virtual ~AutoEnforcerLibraryRegistry(){};
```

Comment:
Missing space before `{}`. Does this pass `check-conventions`?

Reply:
yeah, it had no complaints about it
isn't something that clang format should have caught toh?

Reply:
Yeah, are you using `orc shell ./scripts/check-revng-conventions --force-format`?

Reply:
yes

Reply:
i tries fixing this by hand but force format will remove the space.

i used  = default instead